storage/pruning/fileHandling.go

```diff
@@ -0,0 +1,51 @@
+package pruning
+
+import (
+	"io"
+	"os"
+	"strings"
+)
+
+// removeDirectoryIfEmpty will clean the directories after all persisters for one epoch were destroyed
+// the structure is this way :
+// workspace/db/Epoch_X/Shard_Y/DbName
+// we need to remove the last 3 directories if everything is empty
+func removeDirectoryIfEmpty(path string) {
+	elementsSplitBySeparator := strings.Split(path, string(os.PathSeparator))
+
+	epochDirectory := ""
+	for idx := 0; idx < len(elementsSplitBySeparator)-2; idx++ {
+		epochDirectory += elementsSplitBySeparator[idx] + string(os.PathSeparator)
+	}
+
+	if len(elementsSplitBySeparator) > 2 { // if length is less than 2, the path is incorrect
+		shardDirectory := epochDirectory + elementsSplitBySeparator[len(elementsSplitBySeparator)-2]
+		if isDirectoryEmpty(shardDirectory) {
+			err := os.RemoveAll(shardDirectory)
```

Comment:
I see you are removing the epoch directory if the shard directory is empty, is this correct? isn't it possible we have some other shard data inside? e.g multiple observers (different shards) running on the same machine.

If we anyway remove shard and epoch (which is parent) why do we have to do them separately? wouldn't removing epoch directory remove also all children ?

Reply:
Added a check for empty `Epoch_X` directory and only if this is empty I remove it.